crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1678,15 +1678,13 @@ impl<'db> Type<'db> {
             Type::Any | Type::Todo(_) | Type::Never | Type::Unknown => Truthiness::Ambiguous,
             Type::FunctionLiteral(_) => Truthiness::AlwaysTrue,
             Type::ModuleLiteral(_) => Truthiness::AlwaysTrue,
-            Type::ClassLiteral(_) => {
-                // TODO: lookup `__bool__` and `__len__` methods on the class's metaclass
-                // More info in https://docs.python.org/3/library/stdtypes.html#truth-value-testing
-                Truthiness::Ambiguous
-            }
-            Type::SubclassOf(_) => {
-                // TODO: see above
-                Truthiness::Ambiguous
+            Type::ClassLiteral(ClassLiteralType { class }) => {
+                class.metaclass(db).to_instance(db).bool(db)
```

Comment:
I think all of the tests in this PR exercise only the below case (`SubclassOf`), because that's what a `type[...]` annotation spells; I don't think any of them exercise this `ClassLiteral` case. To do that, we'll want to add some tests directly with the class name itself, rather than `type[..]` annotation.